apiary-metastore-events/kafka-metastore-events/kafka-events-common/src/test/java/com.expediagroup.apiary.extensions.events.metastore/common/metrics/HiveMetricsHelperTest.java

```diff
@@ -16,47 +16,40 @@
 package com.expediagroup.apiary.extensions.events.metastore.common.metrics;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.mockito.Mockito.when;
-import static org.powermock.api.mockito.PowerMockito.mockStatic;
 
-import org.apache.hadoop.hive.common.metrics.common.Metrics;
 import org.apache.hadoop.hive.common.metrics.common.MetricsFactory;
+import org.apache.hadoop.hive.conf.HiveConf;
 import org.junit.Before;
 import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.mockito.Mock;
-import org.powermock.core.classloader.annotations.PrepareForTest;
-import org.powermock.modules.junit4.PowerMockRunner;
 
-@RunWith(PowerMockRunner.class)
-@PrepareForTest(MetricsFactory.class)
 public class HiveMetricsHelperTest {
 
-  private @Mock Metrics metrics;
+  private HiveConf conf = new HiveConf();
 
   @Before
-  public void init() throws Exception {
-    mockStatic(MetricsFactory.class);
+  public void setup() throws Exception {
+    MetricsFactory.close();
   }
 
   @Test
-  public void nullMetricsFactory() {
-    when(MetricsFactory.getInstance()).thenReturn(null);
-    assertThat(HiveMetricsHelper.incrementCounter("name")).isNotPresent();
+  public void incrementCounter() throws Exception {
+    conf.setVar(HiveConf.ConfVars.HIVE_METRICS_CLASS, "org.apache.hadoop.hive.common.metrics.metrics2.CodahaleMetrics");
+    MetricsFactory.init(conf);
+    assertThat(HiveMetricsHelper.incrementCounter("name")).get().isEqualTo(1L);
   }
 
   @Test
-  public void exceptionInIncrementCounter() {
-    when(MetricsFactory.getInstance()).thenReturn(metrics);
-    when(metrics.incrementCounter("name")).thenThrow(RuntimeException.class);
+  public void nullMetricsClass() {
+    conf.setVar(HiveConf.ConfVars.HIVE_METRICS_CLASS, "org.apache.hadoop.hive.common.metrics.nullMetrics");
     assertThat(HiveMetricsHelper.incrementCounter("name")).isNotPresent();
```

Comment:
This happens because you don't call `MetricsFactory.init(conf)` right? i.e. it would fail the same way even if you omitted line 43?

Reply:
Yes very true, will remove